### ISSUE TITLE
Gifs/mp4s doesn't show up on my (Samsung)  gallery. 

### ISSUE THREAD

**Issue by RedCapBuffalo** [ccrama/Slide#2688]
When I save a gif/mp4 from Slide, it doesn't show up on my gallery (Samsung). Only after renaming the file does it show up. 
Photos do show up without any issue. 
Slide v5.7.3 doesn't have the issue. 

Slide version: 5.8.2
Android version: 7

**Comment by ccrama** [comment]
@PiwwowPants  could this be related to the new provider in v5.8?

**Comment by RedCapBuffalo** [comment]
Yep. I just installed v5.8 and it too has the same issue. The mp4 file is downloaded and can be found via a file manager but it doesn't show up on my gallery. Only after renaming it shows up.

**Comment by PiwwowPants** [comment]
@ccrama Could be related but I don't see this issue on my end. The need to rename the file confuses me. 

@RedCapBuffalo Can you tell us the name of the file with the extension before you rename it and after you rename it?

**Comment by RedCapBuffalo** [comment]
@ccrama @PiwwowPants 

The issue is with every gifs. Anyways, for the sake of this conversation let's take this post an example-
https://redd.it/85xvdv

Here's the original name of the file - 31e7331d-7bfa-4ea6-ad9a-fda8ff68df51.mp4

I gave it a random name.

Here's the video of the issue: 
https://s.amsu.ng/UnpzzwQRB68N
(Expires: 22 Mar 2018)

**Comment by RedCapBuffalo** [comment]
Another issue that I've found is..... I cannot download a gif/mp4 twice. Even if I delete the file. I can download it only once.
No issues with photos.

**Comment by ccrama** [comment]
What happens if you click the notification to open the video? It also may be that the Samsung gallery doesn't check other folder locations, where do you have gifs set to save in Slide?

**Comment by RedCapBuffalo** [comment]
When I click on the notification, the video plays fine. But it still doesn't show up in the gallery.
And the photos and videos are saved in the internal memory.
Like I said, when I use v5.7 and save the gifs, it does show up without any issue. The issue is only with v5.8. 
So it has nothing to do with the location where the files are saved. Moreover, photos that are saved on the same folder do show up immediately.

**Comment by ccrama** [comment]
What app does it show the video in from the notification? Did your phone happen to update around the same time as Slide did by chance? Just trying to test out all options

**Comment by RedCapBuffalo** [comment]
I can open the file with any media player of my choice. I haven't set a default player.
And yea my phone did get a security update on 9th March. It might be because of that. I think the issue started right around that time.
I tried Slide inside Secure Folder and the issue is present even there.

**Comment by RedCapBuffalo** [comment]
The whole issue is with the filename.
I saved the following gif as mp4 via Slide and Sync. The file that I saved via Slide didn't show up. But the one saved via Sync showed up without any issue.

Gif: https://redd.it/866hso

Slide filename: 2ba47b62-aa20-4f46-bde1-c370af426048.mp4

Sync filename: i9hmGti.mp4

If you make the required changes so that Slide too save the file with a shorter filename, we won't the issue anymore.

**Comment by ccrama** [comment]
Seems like an issue with the Samsung gallery app rather than Slide, but I can look into a different naming convention if that does indeed fix the issue on your device. Although nothing has changed with that for over a year

**Comment by RedCapBuffalo** [comment]
Yea. Give it a shot. I just went back to v5.7.3 and everything works as it should... with the very same name format.
It's so frustrating. I guess I'll give it another try if you change the naming format. Or else I'll just stick to v5.7.3 or move on to Sync.

**Comment by ccrama** [comment]
Then that means it's not the name that is the issue. FWIW I have not received any other reports of this behavior, and the only change that could have an affect is consolidating some of the file saving code in v5.8. Does changing the save location do anything for you? Unfortunately I cannot reproduce the bug which makes it much harder to diagnose. What device are you using?

**Comment by RedCapBuffalo** [comment]
Here's a video of both the version of Slide in action. Have a look....

https://s.amsu.ng/nop6GaoM96WN
(Expires: 23 Mar 2018)

**Comment by Tensounder54** [comment]
This issue is resolved in a later version of Slide and needs to be closed. 
---
It should be noted that this issue appears to occur in Slide version 5.9-alpha2 when it is opened in [Island ](https://play.google.com/store/apps/details?id=com.oasisfeng.island&hl=en_US)which is a `Parallel App` app that allows you to clone apps, for parallel running. I believe that we do not provide support for this but I figured that it should be pointed out.

**Comment by Tensounder54** [comment]
Going to close this one but we should keep an eye on this.
